### DISCUSSION TITLE
Referencing Primary Composite Keys

### DISCUSSION THREAD

**Discussion by abdel-ships-it** [prisma/prisma#12547] [Q&A] [answered]
Im trying to reference a composite key as referenced in the archived documentation but I seem to be getting errors.

Documentations https://github.com/prisma/specs/tree/master/schema#referencing-primary-composite-keys

```prisma
model Document {
  projectID String  @default("")
  revision  Int     @default(1)
  blocks    Block[]
  @@id([projectID, revision])
}

model Block {
  id                Int      @id
  documentProjectID String
  documentRevision  Int
}
```

Error on `blocks Block[]`

> Error validating field `blocks` in model `Document`: The relation field `blocks` on Model `Document` is missing an opposite relation field on the model `Block`. Either run `prisma format` or add it manually.

Now upon running `prisma format` the error goes away and the code ends looking like this

```prisma
model Document {

  projectID String  @default("")
  revision  Int     @default(1)
  blocks    Block[]
  @@id([projectID, revision])
}

model Block {
  id                Int      @id
  document          Document @relation(fields: [documentProjectID, documentRevision], references: [projectID, revision])
  documentProjectID String
  documentRevision  Int
}
```

Ideally I reference the composite primary key as stated in the archived docs, but perhaps thats not possible anymore? Either way I cannot find anything about this in the current docs

**Comment by janpio** [comment, marked as answer]
This is not possible any more. You need to actually have and spell out the fields that contain the referencing fields now. The other solution was much to magic and hard to understand.

**Reply by ooyamatakehisa** [reply]
@janpio So in the latest version, there is no case where we specifcy multiple values in `fields` argument of `@relation()`?

**Reply by janpio** [reply]
No, the opposite is true. You _have_ to spell out the fields if you want to have a relation with multiple fields.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
